target-component-accessor-compiler/ksp/src/main/kotlin/me/tatarka/inject/target/component/accessor/compiler/ksp/TargetComponentAccessorProcessor.kt

```diff
@@ -0,0 +1,146 @@
+package me.tatarka.inject.target.component.accessor.compiler.ksp
+
+import com.google.devtools.ksp.getVisibility
+import com.google.devtools.ksp.processing.CodeGenerator
+import com.google.devtools.ksp.processing.KSPLogger
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.processing.SymbolProcessor
+import com.google.devtools.ksp.processing.SymbolProcessorEnvironment
+import com.google.devtools.ksp.processing.SymbolProcessorProvider
+import com.google.devtools.ksp.symbol.KSAnnotated
+import com.google.devtools.ksp.symbol.KSDeclaration
+import com.google.devtools.ksp.symbol.KSFunctionDeclaration
+import com.google.devtools.ksp.symbol.KSTypeReference
+import com.google.devtools.ksp.symbol.Modifier
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.ksp.toKModifier
+import com.squareup.kotlinpoet.ksp.toTypeName
+import com.squareup.kotlinpoet.ksp.writeTo
+import me.tatarka.inject.annotations.TargetComponentAccessor
+
+class TargetComponentAccessorProcessor(
+    private val codeGenerator: CodeGenerator,
+    private val logger: KSPLogger,
+) : SymbolProcessor {
+    @Suppress("LoopWithTooManyJumpStatements")
+    override fun process(resolver: Resolver): List<KSAnnotated> {
+        val symbols = resolver.getSymbolsWithAnnotation(
+            requireNotNull(TargetComponentAccessor::class.qualifiedName),
+        )
```

Comment:
`getSymbolsWithAnnotation` does a pretty aggressive scan of all classes, methods, and nested classes. Would there be any downside to only scanning top-level functions? You can look at the existing processor to see how this can be done.

Reply:
Yeah that should be fine.